### PR TITLE
Support for checkout service in GET /files

### LINKED ISSUE #1310: GET /files/<uuid> should return links to checkout bucket
blah blah.

### PR DESCRIPTION
**Pull request by ttung**
Depends on #1336, #1343 
Fixes #1310

### CODE REVIEW

**Comment by mweiden** [comment]
@ttung What happened here? Are you splitting this up?

**Comment by ttung** [comment]
Github mistakenly closed this.

File: dss/api/files.py [hunk 1]
```diff
@@ -108,6 +109,39 @@ def get_helper(uuid: str, replica: Replica, version: str=None):
...
+        decoded_token = {
+            CheckoutTokenKeys.EXECUTION_ID: execution_id,
+            CheckoutTokenKeys.START_TIME: start_time,
+            CheckoutTokenKeys.ATTEMPTS: attempts
+        }
+    else:
+        try:
+            decoded_token = json.loads(token)
+            decoded_token[CheckoutTokenKeys.ATTEMPTS] += 1
+        except Exception as ex:
```
Comment (mweiden) [inline dss/api/files.py]: All exceptions? Why not just `(KeyError, ValueError)`?

File: dss/storage/checkout.py [hunk 2]
```diff
@@ -23,6 +26,90 @@ class ValidationEnum(Enum):
...
 
+class BundleNotFoundError(Exception):
+    """Raised when we attempt to check out a non-existent bundle."""
+    pass
+
+
+class CheckoutTokenKeys:
+    """
+    When we are executing a request that involves a checkout, the client will periodically check back in to see if the
+    checkout is complete.  To avoid duplicating checkout requests, the client will check back using a token.  These are
```
Comment (mweiden) [inline dss/storage/checkout.py]: It seems like the deduplication is only done by checking that the token exists, the only key-value pair touched is `attempts`, and that value isn't leveraged. Seems like a simple flag would perform the same function(?) Is there a dev UX element here?

File: dss/api/files.py (hunk 1 above)
Comment (ttung) [reply dss/api/files.py]: Good idea, will fix.

File: dss/storage/checkout.py (hunk 2 above)
Comment (ttung) [reply dss/storage/checkout.py]: It's for future use.  If time has elapsed very little, but attempts is high, that means we have a client that's not respecting the Retry-After header.  We could also implement exponential backoff or other creative solutions for managing retries.

**Pull request merged**